plugin/src/main/java/org/openrewrite/gradle/RewriteDryRunTask.java

```diff
@@ -43,6 +44,7 @@ public RewriteDryRunTask(Configuration configuration, SourceSet sourceSet, Rewri
         super(configuration, sourceSet, extension);
         setGroup("rewrite");
         setDescription("Dry run the active refactoring recipes to sources within the " + sourceSet.getName() + " SourceSet. No results will be made.");
+        getOutputs().upToDateWhen(task -> !extension.getFailOnDryRunResults());
```

Comment:
Maybe better to just have this be `getOutputs().upToDateWhen(Specs. SATISFIES_NONE)` to prevent the output of `dryRun` from ever being cached. That might be simpler, less surprising, and keep functionality in-line with what the maven-plugin will be 🤔

Reply:
Actually, going to invert it. Let's just not cache the result. It's overall less surprising this way.